lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -376,7 +376,7 @@ def write(chunk)
           elsif record.has_key?(@time_key)
             rts = record[@time_key]
             dt = parse_time(rts, time, tag)
-            record[TIMESTAMP_FIELD] = rts unless @time_key_exclude_timestamp
```

Comment:
IMO, specifying `time_key` case should handle timestamp as-is instead of iso8601 conversion.

Reply:
I think that it is expected that the plugin be able to parse timestamps in different formats, and output them in the expected "logstash" format.
Not all logs have their timestamps in the logstash format, and to my understanding thats why time_key_format is for.
Just pushing back the unparsed time field value into `@timestamp` field seems awkward.
This can also cause milliseconds to be lost when inserting the  `@timestamp` field into elasticsearch.

I saw your PR #358 , but still you assume that the input is allready in `iso8601` (elastic) format.
Please reconsider this.

Reply:
Hmm, the above article's claim is reasonable for me.
Thank you for your suggestion. :+1: